src/sqlfluff/core/rules/std/L042.py

```diff
@@ -0,0 +1,64 @@
+"""Implementation of Rule L042."""
+
+from sqlfluff.core.rules.base import BaseCrawler, LintResult
+
+
+class Rule_L042(BaseCrawler):
+    """Join clauses should not contain subqueries. Use CTEs instead.
+
+    NB: Some dialects don't allow CTEs, and for those dialects
```

Comment:
I feel like that should be an issue?

I think the base rule class "BaseCrawler" should have an `exclude_dialects = ` optional property

Reply:
great idea - but in this case it wouldn't help. It's only older versions of mysql which don't support it, newer ones are fine. Do you know of any other rules which might use a feature like this?